### ISSUE TITLE
Multiple rule `ignore` clauses in rules fails producing false positives

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#1460]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where multiple `ignore` clauses in `comment-empty-line-before"` fails producing false positives.

> Which rule, if any, is this issue related to?

`comment-empty-line-before`

> What CSS is needed to reproduce this issue?

```css
/* a */
/* b */
/* c */
.foo {
  color: pink;
}
```

> What stylelint configuration is needed to reproduce this issue?

In `index.js`:
```js
    "comment-empty-line-before": [ "always", {
      ignore: ["between-comments"],
      ignore: ["stylelint-commands"],
    } ],
```
> Which version of stylelint are you using?

`6.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint src/test.css --config index.js`"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
 2:1   ✖  Expected empty line before comment   comment-empty-line-before
 3:1   ✖  Expected empty line before comment   comment-empty-line-before
```

**Comment by Nirazul** [comment]
I don't think that it is valid to add two keys with the same value in JS and JSON. All information I've found points in this direction.

[JS](http://stackoverflow.com/questions/12561482/javascript-how-to-iterate-object-with-two-the-same-keys-and-get-two-values)

[JSON](http://stackoverflow.com/questions/5306741/do-json-keys-need-to-be-unique)

**Comment by ntwb** [comment]
A JS configuration workaround is to reverse the `ignore` clauses configuration:
```js
"comment-empty-line-before": [ "always", {
      ignore: ["stylelint-commands"],
      ignore: ["between-comments"],
    } ],
```
This results in no errors

----

Another JS configuration:
```js
    "comment-empty-line-before": [ "always", {
      ignore: ["between-comments"],
      except: ["first-nested"],
    } ],
```
This results in no errors

----

A  `.stylelintrc` file configuration
```json
  "comment-empty-line-before": [ "always", {
    except: ["first-nested"],
    ignore: ["between-comments"],
  } ],
```
This results in no errors

----

Another `.stylelintrc` file configuration

```json
  "comment-empty-line-before": [ "always", {
    ignore: ["between-comments"],
    ignore: ["stylelint-commands"],
  } ],
```

Running `stylelint test.css --config .stylelintrc` results in:
```shell
Error: Failed to parse "/Users/netweb/dev/stylelint/bug-1460/.stylelintrc" as JSON, JS, or YAML.
    at /usr/local/lib/node_modules/stylelint/node_modules/cosmiconfig/lib/loadDefinedFile.js:32:13
```

This looks like an issue where multiple `ignore` clauses for a rule do not work.

If this is the case it could be either a stylelint issue or a cosmiconfig issue

Also expecting other rules that include options for multiple `ignore`, `exclude` rule options

**Comment by Nirazul** [comment]
Adding this here as your second comment came directly after me responding :)

I don't think that it is valid to add two keys with the same value in JS and JSON. All information I've found points in this direction.

[Is it valid in JS?](http://stackoverflow.com/questions/12561482/javascript-how-to-iterate-object-with-two-the-same-keys-and-get-two-values)
[Is it valid in JSON?](http://stackoverflow.com/questions/5306741/do-json-keys-need-to-be-unique)

**Comment by ntwb** [comment]
> Adding this here as your second comment came directly after me responding :)

Haha 😬 

> I don't think that it is valid to add two keys with the same value in JS and JSON. All information I've found points in this direction.

Yes, thats what appears to the case, thanks for the link references @Nirazul

**Comment by jeddy3** [comment]
> Also expecting other rules that include options for multiple ignore, exclude rule options

Both the `ignore` and `except` options accept *an array of keywords*. Like so:

```json
{ 
  "comment-empty-line-before": [ "always", {
      ignore: ["stylelint-commands", "between-comments"],
    } ],
}
```

Can you try that please?

**Comment by ntwb** [comment]
Thanks @jeddy3, that indeed works, I also had a quick read of some of the docs I'd expect to see an example of this and I didn't find one, so I'd say without a _code search_ I wouldn't find the docs you're referring to, I would have expected the "rules" example here http://stylelint.io/user-guide/configuration/#rules to maybe have showed that?

Also via http://stylelint.io/  `Options validator: So that you can be confident that your config is valid.`, my 2nd comment above has 4 configurations I tested, the 4th with a `.stylelintrc` let me know the configuration wasn't valid via the message `Error: Failed to parse...`, the 1st example using a JavaScript configuration gave no warnings of an invalid config, should it have and was this the correct message I should have seen for both invalid configurations?

**Comment by jeddy3** [comment]
> the 1st example using a JavaScript configuration gave no warnings of an invalid config, should it have and was this the correct message I should have seen for both invalid configurations?

Yep, I think it should have. Perhaps the options validator can be shored-up to catch this.

> I would have expected the "rules" example here http://stylelint.io/user-guide/configuration/#rules to maybe have showed that?

We can update that example and add an `ignore` option with two keywords.

> I also had a quick read of some of the docs I'd expect to see an example of this and I didn't find one

I meant the rule's [README](https://github.com/stylelint/stylelint/blob/master/src/rules/comment-empty-line-before/README.md#optional-options). I think we can definitely make the use of the `except` and `ignore` options a bit clearer. Perhaps if we make the keywords subheadings below an `except` and an `ignore` heading?

Do you think those three changes would resolve this?

**Comment by ntwb** [comment]
> We can update that example and add an ignore option with two keywords.

SGTM 👍 

How about the below example adds the only two rules with multiple `ignore` and `except` options, `comment-empty-line-before` and `rule-nested-empty-line-before`

```js
{
  "rules": {
    "block-no-empty": null,
    "color-no-invalid-hex": true,
    "comment-empty-line-before": [ "always", {
      ignore: ["stylelint-commands", "between-comments"],
    } ],
    "declaration-colon-space-after": "always",
    "indentation": ["tab", {
      "except": ["value"]
    }],
    "max-empty-lines": 2,
    "rule-nested-empty-line-before": [ "always", {
      except: ["after-comment", "first-nested"],
    } ],
    "unit-whitelist": ["em", "rem", "%", "s"]
  }
}
```

> I meant the rule's README. I think we can definitely make the use of the `except` and `ignore` options a bit clearer. Perhaps if we make the keywords subheadings below an `except` and an `ignore` heading?

Yes, I think this would for sure help clarify things 👍

**Comment by jeddy3** [comment]
I really like the updated example. 

I've pushed the other two things into separate issues. This one is just for the example update.

**Comment by ntwb** [comment]
Reopening, the `rule-nested-empty-line-before` rule example should be:
```js
    "rule-nested-empty-line-before": [ "always", {
      "except": ["first-nested"],
      "ignore": ["after-comment"],
    } ],
```

And not:
```js
    "rule-nested-empty-line-before": [ "always", {
      "except": ["after-comment", "first-nested"],
    } ],
```

i.e. the `"after-comment"` option is an `ignore` option not an `except`option
